### PR TITLE
Moved to a Phrased based Data model

### PR DESCRIPTION
**Pull request by Rahgooy**
Moved to a Phrased based Data model

### CODE REVIEW

**Comment by kordjamshidi** [comment]
Thanks @Rahgooy , but don't we want to keep the token-based and just add the phrase-based? what do you think?

**Comment by Rahgooy** [comment]
I wanted to that first, but it means to duplicate almost everything in the code, one for phrase and one for token.

**Comment by kordjamshidi** [comment]
yes, right but what is the problem with that. those are different types and we have to repeat if we want to keep both. But if it is difficult now to have that, it is ok.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala [hunk 1]
```diff
@@ -184,95 +194,95 @@ object MultiModalSpRLDataModel extends DataModel {
...
       val (first, second) = getArguments(r)
       pos(first) + "::" + pos(second)
   }
 
-  val relationSemanticRole = property(pairs) {
+  val relationSemanticRole = property(pairs, cache = true) {
     r: Relation =>
       val (first, second) = getArguments(r)
       semanticRole(first) + "::" + semanticRole(second)
   }
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala:213]: so you managed to run  it wih semantic roles?

Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala:213]: no, the `samnticRole` property return an empty string for now.

Comment (kordjamshidi) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala:213]: ah , ok.

**Comment by kordjamshidi** [comment]
This is good to me to merge.

**Pull request merged**